src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -163,6 +167,36 @@ export class DynamicFacet extends Component implements IAutoLayoutAdjustableInsi
      */
     enableCollapse: ComponentOptions.buildBooleanOption({ defaultValue: false, section: 'Filtering' }),
 
+    /**
+     * Whether to allow the end-user to search the facet values.
+     *
+     * **Default:** `false`
+     */
+    enableFacetSearch: ComponentOptions.buildBooleanOption({ defaultValue: false, section: 'Filtering' }),
+
+    /**
+     * If the [`enableFacetSearch`]{@link DynamicFacet.options.enableFacetSearch} option is `true`,
+     * specifies whether to use a leading wildcard in the facet search query.
+     *
+     * **Default:** `false`
+     */
+    useLeadingWildcardInFacetSearch: ComponentOptions.buildBooleanOption({
+      defaultValue: false,
+      section: 'Filtering',
+      depend: 'enableFacetSearch'
+    }),
+
+    /**
+     * If the [`enableFacetSearch`]{@link DynamicFacet.options.enableFacetSearch} option is `true`, specifies the delay (in
+     * milliseconds) before sending a search request to the server when the user starts typing in the facet search box.
+     *
+     * Specifying a smaller value makes results appear faster. However, chances of having to cancel many requests
+     * sent to the server increase as the user keeps on typing new characters.
+     *
+     * Default value is `400`. Minimum value is `0`.
+     */
+    facetSearchDelay: ComponentOptions.buildNumberOption({ defaultValue: 400, min: 0, section: 'Filtering', depend: 'enableFacetSearch' }),
```

Comment:
I would not make this configurable. 
It's a bit overkill as far as configuration option goes.